sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -69,6 +73,20 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
      * this will be the tracing provider specified in the application context.
      */
     std::shared_ptr<Azure::Core::Tracing::TracerProvider> TracingProvider;
+
+  private:
+    friend class _internal::TelemetryPolicy;
+    long CppStandardVersion =
+#if defined(_azure_BUILDING_SDK)
+        -2L
+#elif defined(_azure_BUILDING_TESTS)
+        -1L
+#elif defined(_azure_BUILDING_SAMPLES)
+        0L
+#else
```

Comment:
What's the significance of these values, in particular. Consider a comment for why we are doing this.

Reply:
Rather than a comment, possibly define 3 manifest constants for each of the three cases you're calling out here. I suspect that would be sufficient to explain what's going on.

Also, do we care if we're building tests AND samples?

Reply:
Each target is either a test or a sample, so there should not be a case when both are active.